qcodes/instrument/visa.py

```diff
@@ -258,3 +259,21 @@ def snapshot_base(self, update: bool = True,
         snap['timeout'] = self.timeout.get()
 
         return snap
+
+    def wait_until_complete(self, check_after: float, sleep_time: float) -> None:
+        """
+        A utility function that asynchronously queries the Event Status Register (ESR)
+        and checks if the ``0`` bit (the Operation Complete Query (OPC) bit) is set, i.e.,
+        the operation is complete. This function repeatedly returns the content of the ESR
+        and clears it.
+
+        Args:
+            check_after: Time to wait after which polling the OPC status to ESR starts.
+            sleep_time: Time to sleep at each iteration in the polling loop.
+        """
+        log.debug("Clearing ESR")
+        self.ask_raw("*ESR?")
```

Comment:
shouldn't this be `*CLS` as described in https://cdn.rohde-schwarz.com/pws/dl_downloads/dl_application/application_notes/1gp79/1GP79_1E_SCPI_Programming_Guide_SigGens.pdf ? i don't know, just trying to sanity-check.

Reply:
I actually read it. If you read section 4.2.2.1, this PR just Pythonic implementation of it. There, CLS is used for initial clean up, can be implemented here as well, I am not sure.

Reply:
aha, i see. indeed, we should not need "initial clean up" here.

but why do we need this initial `*ESR?` call then? the document does not mention it. it just says that one should call `*ESR?` in order to poll only.

Reply:
Ok, I believe  ``*ESR?`` and ``*CLS`` is inter-changeable here, as both clears the ESR. The advantage of ``*ESR?`` is it also returns the content if there is any.

Reply:
ok, good then.